controllers/provider-gcp/pkg/apis/gcp/v1alpha1/types_infrastructure.go

```diff
@@ -109,3 +111,13 @@ type CloudNAT struct {
 	// +optional
 	MinPortsPerVM *int32 `json:"minPortsPerVM,omitempty"`
 }
+
+// FlowLog contains the configuration options for the vpc flow logs
+type FlowLog struct {
+	// AggregationInterval for collecting flow logs
+	AggregationInterval string `json:"aggregationInterval,omitempty"`
+	// FlowSampling sets the sampling rate of VPC flow logs within the subnetwork where 1.0 means all collected logs are reported and 0.0 means no logs are reported
+	FlowSampling *float32 `json:"flowSampling,omitempty"`
+	// Metadata configures whether metadata fields should be added to the reported VPC flow logs
+	Metadata string `json:"metadata,omitempty"`
```

Comment:
Is this an optional or a required field? If optional, then it must be a `*string` and `// +optional` tag should be there. If required, then remove `omitempty`.